### ISSUE TITLE
header functionality broken with 4.6.3?

### ISSUE THREAD

**Issue by phil-f** [mholt/PapaParse#617]
The following code works fine with v4.6.2:

```
processFile = async (csvFile) => {
    Papa.parse(csvFile, {
      skipEmptyLines: true,
      header: true,
      complete: async results => {
          //results.data shows as a correctly formatted JSON object 
}}
```

This same code with v4.6.3 doesn't seem to work at all, instead of producing an object with multiple properties corresponding to the header values, it produces an object with one property containing all the header values.

**Comment by pokoli** [comment]
Could you provide some minimal test case to reproduce the issue?

**Comment by astik** [comment]
Hi all,
Similar problem here, tab-separated CSV seems to be identified with a comma separator with 4.6.3. It is working as expected in 4.6.2.
If i can find time for it, I'll try to add a test case to existing test suite.

**Comment by pokoli** [comment]
Probably this is due to https://github.com/mholt/PapaParse/commit/8af1d23646ecf15ccd7cdbf6d84966e3ebdc6fa0 or https://github.com/mholt/PapaParse/commit/886fdf3cb29b87979a2b0d7ed722f9b89a37e215 which are the only new changes on 4.6.3

**Comment by pokoli** [comment]
But the behaviour should be also reproducible using the master branch as both commits are also present there

**Comment by zgriesinger** [comment]
I don't have a test case to show at the moment, but I can confirm this bug happens when parsing a comma separated CSV, it looks like it has to do with how PapaParse is guessing the delimiter.

For instance, this will incorrectly make: `results.meta.fields` an array with one item, all of the headers in one string.
```javascript
Papa.parse(data, {
    header: true,
    ...
})
```
But this works fine:
```javascript
Papa.parse(data, {
    header: true,
    delimiter: ','
    ...
})
```

**Comment by jbelien** [comment]
Same issue here !

The following CSV file is parsed correctly in `v4.6.2` but not in `v4.6.3`.

```csv
WKT|id|label|color
POINT (30 10)|1|test-1|red
POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10),(20 30, 35 35, 30 20, 20 30))|2|test-2|
```

Options are 
```js
{
        dynamicTyping: true,
        download: true,
        header: true,
        skipEmptyLines: true,
}
```

**Comment by pokoli** [comment]
Could you test this is fixed on current master? #687 may have fixed this.

**Comment by astik** [comment]
Hi all,
I just tested the master branch (commit 49170b7) and it is working with tab delimited CSV file.
It was working with 4.6.2 but not working with 4.6.3 and 5.0.0.
I'm glad that a solution has been found =)

**Comment by pokoli** [comment]
@astik Thanks for your feedback. I'm closing this issue. 

BTW: I've just released 5.0.1 which contains the fix.

**Comment by jbelien** [comment]
I confirm that this issue is fixed in version `5.0.1`! 🎉
